openverse_catalog/dags/common/storage/media.py

```diff
@@ -111,13 +111,24 @@ def clean_media_metadata(self, **media_data) -> dict | None:
         - add `provider`,
         - add default `category`, if available.
 
-        Returns None if license is invalid
+        Returns None if license is invalid, or if missing required `foreign_identifier`,
+        `foreign_landing_url`, or `url`.
         """
         if media_data["license_info"].license is None or not is_valid_license_info(
             media_data["license_info"]
         ):
             logger.debug("Discarding media due to invalid license")
             return None
+
+        for field in [
+            "foreign_identifier",
+            "foreign_landing_url",
+            f"{self.media_type}_url",
+        ]:
+            if media_data.get(field) is None:
+                logger.debug(f"Discarding media due to missing {field}")
+                return None
```

Comment:
I can't believe this wasn't checked before 😱 Well noticed. However, I wonder whether just ignoring items when missing fields at this point, in the media store class, is the right approach. I am slightly inclined to raise an exception here and have each provider DAG handle theses cases explicitly. Sometimes, if a field is not found another from the source can be used in replacement, e.g. this is common in the `creator_url`. These errors have also helped me realize some gaps in the logic in a few cases!

What do you think?

Reply:
> Sometimes, if a field is not found another from the source can be used in replacement, e.g. this is common in the creator_url

That's a really great point -- raising an exception here might help us spot these opportunities in some cases, and even in cases where the record should ultimately be discarded, it'll be an easy fix in the provider class. I'll update!